courier/courier.go

```diff
@@ -45,26 +45,31 @@ func NewSMTP(d smtpDependencies, c *config.Config) *Courier {
 		Port:     int(port),
 		Username: uri.User.Username(),
 		Password: password,
-		// We are setting this to false because it breaks STARTTLS which is the most
-		// common SMTP auto today. SSL is almost never used.
-		SSL:          false,
+
 		Timeout:      time.Second * 10,
 		RetryFailure: true,
 	}
 
-	//var ssl bool
-	//var tlsConfig *tls.Config
-	if uri.Scheme == "smtps" {
-		sslSkipVerify, _ := strconv.ParseBool(uri.Query().Get("skip_ssl_verify"))
+	sslSkipVerify, _ := strconv.ParseBool(uri.Query().Get("skip_ssl_verify"))
+
+	// SMTP schemes
+	// smtp: smtp clear text (with uri parameter) or with StartTLS (enforced by default)
+	// smtps: smtp with implicit TLS (recommended way in 2021 to avoid StartTLS downgrade attacks
+	//    and defaulting to fully-encrypted protocols https://datatracker.ietf.org/doc/html/rfc8314)
+	switch uri.Scheme {
+	case "smtp":
+		// Enforcing StartTLS by default for security best practices (config review, etc.)
+		skipStartTLS, _ := strconv.ParseBool(uri.Query().Get("skip_starttls"))
```

Comment:
I think this should be `disable_starttls` as we're not really skipping it but rather not using it. WDYT?

Reply:
Agreed. I was proposing skip_starttls at first to convince you using a similar patern as skip_ssl_verify.

I'll do the changes.